maintenance/projects/js-toolkit/packages/liferay-npm-build-support/src/loader/normalize-angular-output.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+import {
+	BundlerLoaderContext,
+	BundlerLoaderReturn,
+} from 'liferay-npm-build-tools-common/lib/api/loaders';
+
+/**
+ * This loader:
+ *
+ *   1. Discards any webpack artifact not containing ES5 code
+ *   2. Removes `-es5` from file and module names when present
+ *
+ * This way, the bundler can treat any known Angular version the same, no matter
+ * how webpack produced its output.
+ *
+ * This is because Angular version <=10 produce artifacts for ES5 and ES2015
+ * targets, while Angular versions >=10 only produce one type of artifact, not
+ * differentiated by file name.
+ */
+export default function (context: BundlerLoaderContext): BundlerLoaderReturn {
+	const {filePath, log} = context;
+
+	if (filePath.includes('-es') && !filePath.includes('-es5.')) {
```

Comment:
Is it possible that I could have a file with a substring like `products-essential-oils`? Don't know what the naming conventions in Angular are... Maybe we should tighten this up a bit?

Reply:
:thinking: That's true. But I'm not sure if I can tighten it more because that `-es` is in the middle of the name I think. Let me see what I can do, I may put an AI in place just to make sure this doesn't fail. Otherwise it will fail one day after the release :roll_eyes: ...

Reply:
It might be as simple as changing `'-es'` to `'-es.'`.

Reply:
OK, it's impossible, because these files have fixed names, they don't depend on the user's project:

```
main-es2015.f7ffa6ef99984376ef2f.js
main-es5.f7ffa6ef99984376ef2f.js
polyfills-es2015.8110071a28231e4fad12.js
polyfills-es5.2ef310e05779620c39a1.js
runtime-es2015.409e6590615fb48d139f.js
runtime-es5.409e6590615fb48d139f.js
styles.3ff695c00d717f2d2a11.css
```

So, they are always `main`, `polyfills` and `runtime`. If they were changed the adaptation wouldn't work because the AMD loader wouldn't find them.